src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -2145,31 +2145,56 @@ private void VisitStatementsWithLocalFunctions(BoundBlock block)
 
         public override BoundNode? VisitLocalFunctionStatement(BoundLocalFunctionStatement localFunc)
         {
-            var oldSymbol = this.CurrentSymbol;
             var localFuncSymbol = localFunc.Symbol;
-            var delegateInvokeMethod = _delegateInvokeMethod;
-            this.CurrentSymbol = localFuncSymbol;
-            _delegateInvokeMethod = null;
-
-            var oldPending = SavePending(); // we do not support branches into a lambda
-
-            var savedState = this.State;
-            var localFunctionState = GetOrCreateLocalFuncUsages(localFuncSymbol);
+            var localFunctionState = GetOrCreateLocalFuncUsages(localFunc.Symbol);
             // The starting state is the top state, but with captured
             // variables set according to Joining the state at all the
             // local function use sites
-            State = TopState();
+            var state = TopState();
             for (int slot = 1; slot < localFunctionState.StartingState.Capacity; slot++)
             {
                 var symbol = variableBySlot[RootSlot(slot)].Symbol;
                 if (Symbol.IsCaptured(symbol, localFunc.Symbol))
                 {
-                    State[slot] = localFunctionState.StartingState[slot];
+                    state[slot] = localFunctionState.StartingState[slot];
                 }
             }
             localFunctionState.Visited = true;
 
-            if (!localFunc.WasCompilerGenerated) EnterParameters(localFuncSymbol.Parameters);
+            AnalyzeLocalFunctionOrLambda(
+                localFunc,
+                localFuncSymbol,
+                state,
+                delegateInvokeMethod: null,
+                useDelegateInvokeParameterTypes: false);
+
+            SetInvalidResult();
+
+            return null;
+        }
+
+        private void AnalyzeLocalFunctionOrLambda(
+            IBoundLambdaOrFunction lambdaOrFunction,
+            MethodSymbol lambdaOrFunctionSymbol,
+            LocalState state,
+            MethodSymbol? delegateInvokeMethod,
+            bool useDelegateInvokeParameterTypes)
+        {
+            var oldSymbol = this.CurrentSymbol;
+            this.CurrentSymbol = lambdaOrFunctionSymbol;
+
+            Debug.Assert(!useDelegateInvokeParameterTypes || delegateInvokeMethod is object);
+            var oldDelegateInvokeMethod = _delegateInvokeMethod;
+            _delegateInvokeMethod = delegateInvokeMethod;
+            var oldUseDelegateInvokeParameterTypes = _useDelegateInvokeParameterTypes;
+            _useDelegateInvokeParameterTypes = useDelegateInvokeParameterTypes;
+
+            var savedState = this.State;
+            this.State = state;
+
+            var oldPending = SavePending(); // we do not support branches into a lambda
```

Comment:
Feel like this comment says "what", can u make it say "why"?

Reply:
I feel the doc comment on SavePending is adequate in explaining why we call it here, therefore will delete the comment.